recipes/systemc/2.3.3/conanfile.py

```diff
@@ -0,0 +1,113 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class SystemcConan(ConanFile):
+    name = "systemc"
+    version = "2.3.3"
+    description = """SystemC is a set of C++ classes and macros which provide
+                     an event-driven simulation interface."""
+    homepage = "https://www.accellera.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("simulation", "modeling", "esl", "tlm")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "disable_async_updates": [True, False],
+        "disable_copyright_msg": [True, False],
+        "disable_virtual_bind": [True, False],
+        "enable_assertions": [True, False],
+        "enable_immediate_self_notifications": [True, False],
+        "enable_pthreads": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "disable_async_updates": False,
+        "disable_copyright_msg": False,
+        "disable_virtual_bind":  False,
+        "enable_assertions": True,
+        "enable_immediate_self_notifications": False,
+        "enable_pthreads": False
+    }
+    generators = "cmake"
+    exports_sources = "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos build not supported")
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("The compilation of SystemC as a "
+                                            "DLL on Windows is currently not "
+                                            "supported")
+
```

Comment:
```suggestion

        if self.settings.os == "Windows" and self.options.enable_pthreads:
            raise ConanInvalidConfiguration("pthreads option is not supported on Windows")
```
https://github.com/accellera-official/systemc/blob/604182509559ae42c34e878f508bae7c18abbbd6/CMakeLists.txt#L529

Reply:
Instead of throwing here, you can also remove the option in `config_options`.

Reply:
Yes, removing will be better